### PR TITLE
Fix a whole lot of typos.

### PR DESCRIPTION
**Pull request by bangerth**
This used a tool that ASPECT has recently integrated into its CI check. It fins a lot of false positives (which I suspect we could address) plus a lot of actual issues that I'm fixing here.

### CODE REVIEW

File: examples/step-37/step-37.cc [hunk 1]
```diff
@@ -566,7 +566,7 @@ namespace Step37
   // slot and a zero entry in all other slots, i.e., we apply the cell-wise
   // differential operator on one unit vector at a time. The inner part
   // invoking FEEvaluation::evaluate, the loop over quadrature points, and
```
Comment (masterleinad) [inline examples/step-37/step-37.cc:568]:
```suggestion
  // invoking FEEvaluation::evaluate(), the loop over quadrature points, and
```

Comment (bangerth) [reply examples/step-37/step-37.cc]: So fixed!

**Review by masterleinad — APPROVED** [review]

**Review by drwells — APPROVED** [review]

File: include/deal.II/grid/grid_generator.h [hunk 2]
```diff
@@ -1726,7 +1726,7 @@ namespace GridGenerator
    * @p grid_generator_function_arguments.
    *
    * The string that supplies the arguments is passed to the function
-   * Patterns::Tools::Convert<TupleTyple>::to_value(), where `TupleType` here is
+   * Patterns::Tools::Convert<TupleType>::to_value(), where `TupleType` here is
```
Comment (bangerth) [inline include/deal.II/grid/grid_generator.h:1729]: This was my favorite :-)

**Comment by bangerth** [comment]
Yes, I agree. Some of them definitely made me chuckle. I think you can still see in some of the commentary the German roots of their authors, and their not-so-perfect command of the English language at the time :-)

**Pull request merged**